### PR TITLE
Add basic database TF. refined in another feature branch.

### PR DESCRIPTION
**Pull request by kunwarsingh1092**
### Description
Added a cloud sql postgres database for the application:
1. Added db tf
2. Added a secret managed through secret manager
3. Added a user to the db.

### CODE REVIEW

File: cloud/gcp/modules/database/main.tf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+resource "google_secret_manager_secret_version" "database_password_version" {
+  secret = google_secret_manager_secret.database_password.id
+  secret_data = random_password.password.result
+}
+
+resource "google_secret_manager_secret_iam_binding" "binding" {
+  secret_id = google_secret_manager_secret.database_password.secret_id
+  role = "roles/secretmanager.secretAccessor"
+  members = [
+    "user:ktoor@google.com",
```
Comment (goldblatt) [inline cloud/gcp/modules/database/main.tf:23]: can you add a todo here to make this non specific to your user!

**Review by goldblatt — CHANGES_REQUESTED** [review]

File: cloud/gcp/modules/database/main.tf (hunk 1 above)
Comment (kunwarsingh1092) [reply cloud/gcp/modules/database/main.tf:23]: I have changed this in the next cl

**Review by goldblatt — APPROVED** [review]

File: cloud/gcp/modules/database/variables.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+variable "tier_type" {
+    type = string
+    description = "tier type to use for the db instance"
+}
+
+variable "terraform_service_account" {
+    type = string
+    description = "service account being used by terraform"
+}
```
Comment (bion) [inline cloud/gcp/modules/database/variables.tf:14]: nit: missing newline at end of file

File: cloud/gcp/modules/database/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+output "connection_name" {
+    value = google_sql_database_instance.civiform_db.connection_name
+}
+
+output "db-ip" {
+    value = google_sql_database_instance.civiform_db.ip_address.0.ip_address
+}
```
Comment (bion) [inline cloud/gcp/modules/database/outputs.tf:7]: nit: missing newline at end of file

File: cloud/gcp/modules/database/main.tf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  }
+}
+
+## TODO(ktoor@google.com): Change this with the service account running the application.
+resource "google_sql_user" "civiform-user" {
+  name     = "ktoor@google.com"
+  instance = google_sql_database_instance.civiform_db.name
+  password = google_secret_manager_secret_version.database_password_version.secret_data
+  type     = "CLOUD_IAM_USER"
+}
```
Comment (bion) [inline cloud/gcp/modules/database/main.tf:47]: nit: missing newline at end of file

File: cloud/gcp/modules/database/main.tf (hunk 1 above)
Comment (bion) [reply cloud/gcp/modules/database/main.tf:23]: your email is hard-coded in https://github.com/seattle-uat/civiform/pull/1881/files as well

**Review by bion — CHANGES_REQUESTED** [review]

File: cloud/gcp/modules/database/main.tf (hunk 1 above)
Comment (kunwarsingh1092) [reply cloud/gcp/modules/database/main.tf:23]: I have changed it in the next cl. Also, this is just for initial testing. Will product-ionize later

File: cloud/gcp/modules/storage/main.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  encryption {
+    default_kms_key_name = google_kms_crypto_key.storage_key.id
+  }
+
+  depends_on = [
+    google_kms_crypto_key_iam_binding.kms_binding
+  ]
+}
+
+## TODO(ktoor@google.com) : Remove user and add service accounts for GKE.
```
Comment (shubha-rajan) [inline cloud/gcp/modules/storage/main.tf:48]: Same here, please file an issue and reference the issue number here.

File: cloud/gcp/modules/storage/main.tf [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+  uniform_bucket_level_access = true
+
+  versioning {
+    enabled = true
+  }
+
+  encryption {
+    default_kms_key_name = google_kms_crypto_key.storage_key.id
+  }
```
Comment (shubha-rajan) [inline cloud/gcp/modules/storage/main.tf:41]: question (blocking): It looks like Cloud Storage offers some level of at-rest/server-side encryption by default. What's the benefit of using our own encryption key here?
https://cloud.google.com/storage/docs/encryption/default-keys

**Review by shubha-rajan — DISMISSED** [review]

File: cloud/gcp/modules/storage/main.tf (hunk 5 above)
Comment (kunwarsingh1092) [reply cloud/gcp/modules/storage/main.tf:48]: Acknowledged

File: cloud/gcp/modules/database/main.tf [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+  settings {
+      ip_configuration {
+      ipv4_enabled    = true
+    }
+    tier = var.tier_type
+  }
+}
+
+## TODO(ktoor@google.com): Change this with the service account running the application.
```
[the remark this answers is not in the document]
Comment (kunwarsingh1092) [reply cloud/gcp/modules/database/main.tf:41]: Acknowledged

File: cloud/gcp/modules/storage/main.tf (hunk 6 above)
Comment (bion) [reply cloud/gcp/modules/storage/main.tf:41]: @kunwarsingh1092 the GCP setup doesn't need to be an exact copy of the AWS environment. If using default encryption is also secure and also easier let's do that.

File: cloud/gcp/modules/database/variables.tf (hunk 2 above)
Comment (kunwarsingh1092) [reply cloud/gcp/modules/database/variables.tf]: done

File: cloud/gcp/modules/database/outputs.tf (hunk 3 above)
Comment (kunwarsingh1092) [reply cloud/gcp/modules/database/outputs.tf]: done

File: cloud/gcp/modules/database/main.tf (hunk 4 above)
Comment (kunwarsingh1092) [reply cloud/gcp/modules/database/main.tf]: done

**Review by bion — APPROVED** [review]

**Review by goldblatt — APPROVED** [review]

**Pull request merged**